GSA/digital-gov-drupal
# DIGITAL-45: Added roles and perms

## Jira ticket

[DIGITAL-45](https://cm-jira.usa.gov/browse/DIGITAL-45)

## Purpose

Added permissions required for the user admin and all editorial roles. The data analyst role does not have their modules installed yet, cannot set perms yet.
I also re-exported all content as published and exported test users.
Added a few modules to make the workflow easier:
* disable_user_1_edit: The user admin has 'administer users'  permission, but they should not be able to edit user 1.
* override_node_options: Needed to give editorial roles the ability to set the authored by and authored on fields without giving them the 'administer content' role.
* remove_username: Email field now becomes the user name field, we don't need the user name field.
* vbo_workflow_transition: Allow editorial roles to change workflow states from the content overview and moderated content views.
* views_bulk_operations: A requirement of vbo_workflow_transition.

**KNOWN ISSUE**: Non-admin users cannot reference un-published references. This also includes using the 'Add node' button.

## Includes the following PRs that must be merged first

https://github.com/GSA/digital-gov-drupal/pull/32

## Deployment and testing

### Local Setup

`lando be && lando si` is required to get the new deps, configuration, and content.

### QA/Testing instructions

Login as the testing users and ensure the transitions can made per the 5th column of https://cm-jira.usa.gov/browse/DIGITAL-34. 
Review the permissions outlined in the 'Background' portion of https://cm-jira.usa.gov/browse/DIGITAL-45 and ensure they are all enforce.
The redirect & aliases are not editible yet because this functionality is not merged.

## Checklist for the Developer

- [] Automated tests have passed on this PR.
- [] A reviewer has been designated.

## Checklist for the Peer Reviewers

- [] The file changes are relevant to the task objective.
- [] Code is readable and includes appropriate commenting.
- [] Code standards and best practices are followed.
- [] QA/Test steps were successfully completed, if applicable.
- [] Applicable logs are free of errors.
